tools/doc-generator/doc-generator.go

```diff
@@ -68,185 +67,42 @@ func main() {
 }
 
-func (m metricList) writeToFile(newFile io.WriteCloser) {
-	for _, met := range m {
-		met.writeToFile(newFile)
-	}
-}
-
-func getMetricsNotIncludeInEndpointByDefault() metricList {
-	metrics := metricList{
-		{
-			name:        domainstats.MigrateVmiDataProcessedMetricName,
-			description: "The total Guest OS data processed and migrated to the new VM.",
-			mType:       "Gauge",
-		},
-		{
-			name:        domainstats.MigrateVmiDataRemainingMetricName,
-			description: "The remaining guest OS data to be migrated to the new VM.",
-			mType:       "Gauge",
-		},
-		{
-			name:        domainstats.MigrateVmiDirtyMemoryRateMetricName,
-			description: "The rate of memory being dirty in the Guest OS.",
-			mType:       "Gauge",
-		},
-		{
-			name:        domainstats.MigrateVmiMemoryTransferRateMetricName,
-			description: "The rate at which the memory is being transferred.",
-			mType:       "Gauge",
-		},
-		{
-			name:        "kubevirt_vmi_phase_count",
-			description: "Sum of VMIs per phase and node. `phase` can be one of the following: [`Pending`, `Scheduling`, `Scheduled`, `Running`, `Succeeded`, `Failed`, `Unknown`].",
-			mType:       "Gauge",
-		},
-		{
-			name:        "kubevirt_vmi_non_evictable",
-			description: "Indication for a VirtualMachine that its eviction strategy is set to Live Migration but is not migratable.",
-			mType:       "Gauge",
-		},
-	}
-
+func getMetrics() []operatormetrics.Metric {
 	err := virt_controller.SetupMetrics(nil, nil, nil, nil, nil, nil, nil, nil)
 	checkError(err)
-	for _, m := range virt_controller.ListMetrics() {
-		metrics = append(metrics, newMetric(m))
-	}
 
 	err = virt_api.SetupMetrics()
 	checkError(err)
-	for _, m := range virt_api.ListMetrics() {
-		metrics = append(metrics, newMetric(m))
-	}
 
 	err = virt_operator.SetupMetrics()
 	checkError(err)
-	for _, m := range virt_operator.ListMetrics() {
-		metrics = append(metrics, newMetric(m))
-	}
-
-	err = rules.SetupRules("")
-	checkError(err)
-
-	for _, rule := range rules.ListRecordingRules() {
-		metrics = append(metrics, metric{
-			name:        rule.GetOpts().Name,
-			description: rule.GetOpts().Help,
-			mType:       string(rule.GetType()),
-		})
-	}
-
-	return metrics
-}
-
-func newMetric(om operatormetrics.Metric) metric {
-	return metric{
-		name:        om.GetOpts().Name,
-		description: om.GetOpts().Help,
-		mType:       strings.Replace(string(om.GetType()), "Vec", "", 1),
-	}
-}
-
-func parseMetricDesc(line string) (string, string) {
-	split := strings.Split(line, " ")
-	name := split[2]
-	split[3] = strings.Title(split[3])
-	description := strings.Join(split[3:], " ")
-	return name, description
-}
 
-func parseMetricType(scan *bufio.Scanner, name string) string {
-	for scan.Scan() {
-		typeLine := scan.Text()
-		if strings.HasPrefix(typeLine, "# TYPE ") {
-			split := strings.Split(typeLine, " ")
-			if split[2] == name {
-				return strings.Title(split[3])
-			}
-		}
-	}
-	return ""
+	return virt_controller.ListMetrics()
```

Comment:
since it's metrics from multiple components, you should maybe use `operatormetrics.ListMetrics()` directly